aea/protocols/generator.py

```diff
@@ -18,25 +18,30 @@
 # ------------------------------------------------------------------------------
 """This module contains the protocol generator."""
 
-import argparse
-# import inspect
 import os
 import re
 from os import path
 from pathlib import Path
 import yaml
 from typing import Any, Dict, List, Set
 
-# CUR_PATH = os.path.dirname(inspect.getfile(inspect.currentframe()))
-# PWD = os.path.join(CUR_PATH, "..")
-PWD = ".."
 DEFAULT_TYPES = ["int", "float", "bool", "str", "bytes", "list", "dict", "tuple", "set"]
+
 MESSAGE_IMPORT = "from aea.protocols.base import Message"
 INIT_FILE_NAME = "__init__.py"
 MESSAGE_FILE_NAME = "message.py"
 SERIALIZATION_FILE_NAME = "serialization.py"
 PROTOCOL_FILE_NAME = "protocol.yaml"
 
+BASIC_FIELDS_AND_TYPES = {
+        "name": str,
+        "authors": str,
+        "version": str,
+        "license": str,
+        "url": str,
```

Comment:
We are dropping the url field so it can be removed

Reply:
Keeping URL may not be a bad idea, developers could include a link to RFC, standard, specification webpage for the connection/skill/protocol they developed.

Reply:
We decided to drop it for multiple reasons for now. We will only introduce it once we use it. YAGNI applies here!